### PR TITLE
Update working environment.

### PR DESCRIPTION
**Pull request by Natim**
r? @leplatrem

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ COPY ./requirements.txt /app/requirements.txt
 RUN apt-get -q update \
     && apt-get -q --yes install g++ \
     && pip install --upgrade --no-cache-dir -r requirements.txt \
-         gunicorn nose flake8 \
+         gunicorn nose flake8 mock webtest unittest2 \
```
Comment (leplatrem) [inline Dockerfile]: Just out of curiosity, why do we want dev deps like nose or mock in the Docker instance? (A comment would be valuable IMO)

File: Makefile [hunk 2]
```diff
@@ -1,36 +1,75 @@
...
+
+maintainer-clean: distclean
+	rm -fr local/ .tox/
+
+NAME := tokenserver
+SOURCE := $(shell git config remote.origin.url | sed -e 's|git@|https://|g' | sed -e 's|github.com:|github.com/|g')
+VERSION := $(shell git describe --always --tag)
+COMMIT := $(shell git log --pretty=format:'%H' -n 1)
+version-file:
+	echo '{"name":"$(NAME)","version":"$(VERSION)","source":"$(SOURCE)","commit":"$(COMMIT)"}' > version.json
```
Comment (leplatrem) [inline Makefile]: 👍

File: Dockerfile (hunk 1 above)
Comment (leplatrem) [inline Dockerfile]: And why not use `-r dev-requirements.txt` then ?

File: dev-requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+flake8
+nose
+webtest
```
Comment (leplatrem) [inline dev-requirements.txt]: mock?

File: setup.py [hunk 4]
```diff
@@ -4,7 +4,24 @@
...
+requires = [
+    'cornice < 2',
+    'mozsvc >= 0.8',
+    'PyBrowserID',
+    'SQLAlchemy',
+    'boto',
+    'hawkauthlib',
+    'testfixtures',
+    'tokenlib',
+    'umemcache',
```
Comment (leplatrem) [inline setup.py]: Out of curiosity, why do we have so many new dependencies?

File: Dockerfile (hunk 1 above)
Comment (Natim) [inline Dockerfile]: +1 to use dev-requirements.txt now that it exists.
Apparently they want to run tests from the Docker and it is the only place to install the dependencies in it.

File: setup.py (hunk 4 above)
Comment (Natim) [inline setup.py]: They are not new they were only in the requirement.txt file before.

**Pull request merged**